CMakeLists.txt

```diff
@@ -284,23 +304,13 @@ target_link_libraries(hbaseclient-shared ${CMAKE_DL_LIBS})
 add_executable(simple-client "${HBASE_SRC_DIR}/examples/simple-client.cc")
 set_target_properties(simple-client PROPERTIES COMPILE_FLAGS "  ")
 set_target_properties(simple-client PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/bin/examples")
-target_link_libraries(simple-client ${PROTOBUF_LIBRARY})
-target_link_libraries(simple-client ${Boost_LIBRARIES})
-target_link_libraries(simple-client ${SASL_LIBS})
-target_link_libraries(simple-client ${GFLAGS_SHARED_LIB})
-target_link_libraries(simple-client ${KRB5_LIBRARIES})
-target_link_libraries(simple-client ${ZOOKEEPER_LIBRARIES})
 target_link_libraries(simple-client hbaseclient-static ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(simple-client ${WHOLE_ARCHIVE} ${Boost_context} ${NO_WHOLE_ARCHIVE})
```

Comment:
Whats the rationale behind this change?

Reply:
I could not get this to build because of missing symbols in folly. It's inevitably a linking issue somewhere. make_fcontext was being dropped and I believe this is written in assembler and linked differently along their build process. My guess is that we have some linking issues to look into but I'm not sure where those are. My hope was to deal with it by ensuring all symbols were provided from boost and create a follow on to investigate.

Reply:
Cool, fine to do a follow on patch.